new algo

SanOtaku09 opened the pull request:
### **Describe your change:**
two new algo in com.thealgorithm.strings
1. longest non repeative substing
2. zig zag pattern

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

#### References

<!-- Add any reference to previous pull-request or issue -->

### **Checklist:**

<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

siriak reviewed (changes requested):
Please add tests, format code as in other files in the repo, and give descriptive names to the classes and functions instead of `Solution` and `convert`

SanOtaku09 commented:
> Please add tests, format code as in other files in the repo, and give descriptive names to the classes and functions instead of `Solution` and `convert`

renamed the class name and variable 
added test for both file

siriak reviewed (approved):
Looks good, thanks!
